src/status_im/chat/db.cljs

```diff
@@ -0,0 +1,341 @@
+(ns status-im.chat.db
+  (:require [clojure.set :as set]
+            [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.i18n :as i18n]
+            [status-im.utils.config :as utils.config]
+            [status-im.utils.core :as utils]
+            [status-im.utils.gfycat.core :as gfycat]
+            [status-im.utils.identicon :as identicon]
+            [status-im.utils.datetime :as time]
+            [status-im.contact.db :as contact.db]
+            [taoensso.timbre :as log]))
+
+(defn filter-messages-from-blocked-contacts
+  [messages blocked-contacts]
+  (remove #(blocked-contacts (:from %)) messages))
+
+(defn- set-previous-message-info
+  [stream]
+  (let [{:keys [display-photo? message-type] :as previous-message} (peek stream)]
+    (conj (pop stream) (assoc previous-message
+                              :display-username? (and display-photo?
+                                                      (not= :system-message message-type))
+                              :first-in-group?   true))))
+
+(defn display-photo?
+  [{:keys [outgoing message-type]}]
+  (or (= :system-message message-type)
+      (and (not outgoing)
+           (not (= :user-message message-type)))))
+
+(defn- intersperse-datemark
+  "Reduce step which expects the input list of messages to be sorted by clock value.
+  It makes best effort to group them by day.
+  We cannot sort them by :timestamp, as that represents the clock of the sender
+  and we have no guarantees on the order.
+  We naively and arbitrarly group them assuming that out-of-order timestamps
+  fall in the previous bucket.
+  A sends M1 to B with timestamp 2000-01-01T00:00:00
+  B replies M2 with timestamp    1999-12-31-23:59:59
+  M1 needs to be displayed before M2
+  so we bucket both in 1999-12-31"
+  [{:keys [acc last-timestamp last-datemark]} {:keys [timestamp datemark] :as msg}]
+  (cond (empty? acc)                                     ; initial element
```

Comment:
@yenda i believe we agreed once to put comments before the line instead of in the end of line